src/EventLoop.cpp

```diff
@@ -358,44 +441,78 @@ auto requests_accept_async(
     {
         std::lock_guard<std::mutex> guard { event_loop->m_pending_requests_lock };
         // swap so we can release the lock as quickly as possible
-        event_loop->m_grabbed_requests.swap(
-            event_loop->m_pending_requests);
+        event_loop->m_grabbed_requests.swap(event_loop->m_pending_requests);
```

Comment:
looks like that race condition is going to be a bit more difficult to solve, the lock is held here for such a short scope there's plenty of time while there are requests neither in pending nor accounted for in the active requests count